api/post.go

```diff
@@ -984,3 +985,41 @@ func searchPosts(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Write([]byte(posts.ToJson()))
 }
+
+func GetStarredPosts(c *Context, w http.ResponseWriter, r *http.Request) {
+	posts, err := getStarredPosts(c.Session.UserId)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	w.Write([]byte(posts.ToJson()))
+}
+
+func getStarredPosts(userId string) (*model.PostList, *model.AppError) {
+	pchan := Srv.Store.Preference().GetCategory(userId, model.PREFERENCE_CATEGORY_STARRED_POSTS)
+
+	if presult := <-pchan; presult.Err != nil {
+		return nil, presult.Err
+	} else {
+		preferences := presult.Data.(model.Preferences)
+		ids := make([]string, 0, len(preferences))
+
+		for _, pref := range preferences {
+			if pref.Value == "true" {
+				ids = append(ids, pref.Name)
+			}
+		}
+
+		if len(ids) == 0 {
+			return &model.PostList{}, nil
+		} else {
+			pochan := Srv.Store.Post().GetByIds(ids)
+			if poresult := <-pochan; poresult.Err != nil {
+				return nil, poresult.Err
+			} else {
+				return poresult.Data.(*model.PostList), nil
```

Comment:
I wonder if we should check if all the posts are in the same channelId and they are in the current channel.  Otherwise you might be able to pull a msg you don't have access to?  I guess you would need to know the id first (not easy).